src/Core/Domain/Shop/ValueObject/ShopConstraint.php

```diff
@@ -53,38 +53,40 @@ class ShopConstraint
      * Constraint to get configuration for a specific shop
      *
      * @param int $shopId
+     * @param bool $isStrict
      *
      * @return static
      *
      * @throws ShopException
      */
-    public static function shop(int $shopId): self
+    public static function shop(int $shopId, bool $isStrict = false): self
     {
-        return new static($shopId, null, false);
+        return new static($shopId, null, $isStrict);
     }
 
     /**
      * Constraint to get configuration for a specific shop group
      *
      * @param int $shopGroupId
+     * @param bool $isStrict
      *
      * @return static
      *
      * @throws ShopException
      */
-    public static function shopGroup(int $shopGroupId): self
+    public static function shopGroup(int $shopGroupId, bool $isStrict = false): self
     {
-        return new static(null, $shopGroupId, false);
+        return new static(null, $shopGroupId, $isStrict);
     }
 
     /**
-     * Constraint to get configuration for all shops (the global value)
+     * @param bool $isStrict
```

Comment:
Reintroduce the missing comment
```suggestion
     * Constraint to get configuration for all shops (the global value)
     *
     * @param bool $isStrict
```